### PR TITLE
IfcFastener Rail Import BIMQ 2021-07-21T07:07:50.7790077Z

### PR DESCRIPTION
**Pull request by jmirtsch**
Rail properties for IfcFastener

### CODE REVIEW

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_FastenerRailWeld" UniqueId="27eae412-12dc-d9f9-29a1-1578ad14fd41" ApplicableType="IfcFastener/WELD" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+	<Properties>
+		<DocProperty xsi:nil="true" href="BelongingPanel_2I_S0OrPP7Wvu0OPlABgP" />
+		<DocProperty xsi:nil="true" href="ConstructionCompany_3q5el6dSLCw8hGDPwFONxo" />
+		<DocProperty xsi:nil="true" href="IsWeldedInFactory_0aqg9_hP5xeWnJrydF220" />
+		<DocProperty xsi:nil="true" href="IsLiftingBracket_0Py8wuJJrAwPXDvufAW9rN" />
+		<DocProperty xsi:nil="true" href="RelativePosition_06BnzJB6b9zhMKCUAtPkfJ" />
+		<DocProperty xsi:nil="true" href="TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV" />
+		<DocProperty xsi:nil="true" href="TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF" />
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml:10]: Same discussion as usual regarding TrackEdgeId.

File: IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ConstructionCompany_3q5el6dSLCw8hGDPwFONxo" Name="ConstructionCompany" UniqueId="f4168bc6-9dc5-4ce8-8ad0-359e8f617ef2" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
+	<Enumeration xsi:type="DocPropertyEnumeration" xsi:nil="true" href="PEnum_Created" />
+</DocProperty>
+
```
Comment (larswik) [inline IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml:5]: Mixing description of the product with description of the process (e.g. who performed the process of construction) does not feel entirely clean to me. I think that is sort of shortcutting the structure with IfcProcess, IfcProduct and IfcResource.

File: IFC4x3/Properties/i/IK_Code_2kpG9_rK93LwNz4pCQcRnH/DocProperty.xml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="IK_Code_2kpG9_rK93LwNz4pCQcRnH" Name="IK_Code" UniqueId="aecd027e-d542-4357-a5fd-13331a99bc51" />
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="IK_Code_2kpG9_rK93LwNz4pCQcRnH" Name="IK_Code" UniqueId="aecd027e-d542-4357-a5fd-13331a99bc51" PrimaryDataType="IfcLabel" />
```
Comment (larswik) [inline IFC4x3/Properties/i/IK_Code_2kpG9_rK93LwNz4pCQcRnH/DocProperty.xml:3]: Documentation missing

File: IFC4x3/Properties/i/IsLiftingBracket_0Py8wuJJrAwPXDvufAW9rN/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Is it a connection between two different heights? Yes/no
```
Comment (larswik) [inline IFC4x3/Properties/i/IsLiftingBracket_0Py8wuJJrAwPXDvufAW9rN/Documentation.md:1]: I am no expert, but can the definition be improved?

File: IFC4x3/Properties/r/RelativePosition_06BnzJB6b9zhMKCUAtPkfJ/Documentation.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Indication of whether the welded joint lies to the left or the right in the track edge direction
```
Comment (larswik) [inline IFC4x3/Properties/r/RelativePosition_06BnzJB6b9zhMKCUAtPkfJ/Documentation.md:1]: What is the track edge direction? Would it be better to use alignment direction to refer to something hopefully present in the dataset?

File: IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF" Name="TrackEdgeId" UniqueId="d589a22f-3ab2-47a7-afbe-5488ccad2a8f" PrimaryDataType="IfcIdentifier" />
+
```
Comment (larswik) [inline IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml:3]: Usual discussion regarding TrackEdgeId

File: IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq" Name="EXTERNALCOMPANY" UniqueId="94775e81-492f-4bd3-b790-554b3dd41cb4" />
+
```
Comment (larswik) [inline IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml:3]: Documentation missing. See also other comment regarding ConstructionCompany.

File: IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ" Name="INTERNALSTAFF" UniqueId="834dd354-2fe1-4c5c-b917-18ab8a068c9a" />
+
```
Comment (larswik) [inline IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml:3]: Documentation missing. See also other comment regarding ConstructionCompany.

File: IFC4x3/PropertyConstants/t/TRACKPANEL_0IocCGCbjEbQgMuEL_GjIF.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TRACKPANEL_0IocCGCbjEbQgMuEL_GjIF" Name="TRACKPANEL" UniqueId="12ca6310-325b-4e95-aa96-e0e57e42d48f" />
+
```
Comment (larswik) [inline IFC4x3/PropertyConstants/t/TRACKPANEL_0IocCGCbjEbQgMuEL_GjIF.xml:3]: No documentation, even if it is maybe not entirely necessary here.

File: IFC4x3/PropertyConstants/t/TURNOUTPANEL_1G$HCil_D2dOSmwDaZucA0.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TURNOUTPANEL_1GHCil_D2dOSmwDaZucA0" Name="TURNOUTPANEL" UniqueId="50fd132c-bfe3-429d-8730-e8d923e26280" />
+
```
Comment (larswik) [inline IFC4x3/PropertyConstants/t/TURNOUTPANEL_1G$HCil_D2dOSmwDaZucA0.xml:3]: No documentation, even if it is maybe not entirely necessary here.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_FastenerRailWeld" UniqueId="27eae412-12dc-d9f9-29a1-1578ad14fd41" ApplicableType="IfcFastener/WELD" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+	<Properties>
+		<DocProperty xsi:nil="true" href="BelongingPanel_2I_S0OrPP7Wvu0OPlABgP" />
+		<DocProperty xsi:nil="true" href="ConstructionCompany_3q5el6dSLCw8hGDPwFONxo" />
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml:5]: See earlier comment regarding ConstructionCompany.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_FastenerRailWeld" UniqueId="27eae412-12dc-d9f9-29a1-1578ad14fd41" ApplicableType="IfcFastener/WELD" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+	<Properties>
+		<DocProperty xsi:nil="true" href="BelongingPanel_2I_S0OrPP7Wvu0OPlABgP" />
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml:4]: Can BelongingPanel be inferred from decomposition structure?

File: IFC4x3/Properties/i/IK_Code_2kpG9_rK93LwNz4pCQcRnH/DocProperty.xml (hunk 3 above)
Comment (fhulin) [reply IFC4x3/Properties/i/IK_Code_2kpG9_rK93LwNz4pCQcRnH/DocProperty.xml]: Usual problem. Out of railway Scope.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml (hunk 1 above)
Comment (fhulin) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml]: I have erased it. adb5fdd751ea322bf0649365b080a81b2a458a62

File: IFC4x3/Properties/i/IsLiftingBracket_0Py8wuJJrAwPXDvufAW9rN/Documentation.md (hunk 4 above)
Comment (fhulin) [reply IFC4x3/Properties/i/IsLiftingBracket_0Py8wuJJrAwPXDvufAW9rN/Documentation.md]: I have improved it! adb5fdd751ea322bf0649365b080a81b2a458a62

File: IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml (hunk 6 above)
Comment (fhulin) [reply IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml]: I have erased it. adb5fdd751ea322bf0649365b080a81b2a458a62

File: IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml (hunk 7 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml]: I have erased Enum. adb5fdd751ea322bf0649365b080a81b2a458a62

File: IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml (hunk 8 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml]: I have erased Enum. adb5fdd

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml (hunk 11 above)
Comment (fhulin) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml]: I have erased the property. adb5fdd

File: IFC4x3/Properties/r/RelativePosition_06BnzJB6b9zhMKCUAtPkfJ/Documentation.md (hunk 5 above)
Comment (fhulin) [reply IFC4x3/Properties/r/RelativePosition_06BnzJB6b9zhMKCUAtPkfJ/Documentation.md]: Agree. I modified to a property made by Chi in an other pull request.
094c05c0b0e614242b43f17a2c281266c2dd8b63

File: IFC4x3/PropertyConstants/t/TURNOUTPANEL_1G$HCil_D2dOSmwDaZucA0.xml (hunk 10 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/t/TURNOUTPANEL_1G$HCil_D2dOSmwDaZucA0.xml]: erased. Comming from Element assembly. Commit 6cb18533994026ae118add924a2b65d2f98b8472

File: IFC4x3/PropertyConstants/t/TRACKPANEL_0IocCGCbjEbQgMuEL_GjIF.xml (hunk 9 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/t/TRACKPANEL_0IocCGCbjEbQgMuEL_GjIF.xml]: erased. Comming from Element assembly. Commit 6cb18533994026ae118add924a2b65d2f98b8472

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml (hunk 12 above)
Comment (fhulin) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_FastenerRailWeld/DocPropertySet.xml]: Agree, Can be extracted from element assembly it has been aggregated. I have erased the property.6cb18533994026ae118add924a2b65d2f98b8472

**Comment by fhulin** [comment]
@larswik  Now I think It's ok!

**Review by larswik — APPROVED** [review]

File: IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml]: @czapplitec Could use `IfcOrganization` for this.

File: IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml (hunk 8 above)
Comment (SergejMuhic) [reply IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml]: This enum is still here,

File: IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml (hunk 7 above)
Comment (SergejMuhic) [reply IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml]: This enum is still here.

File: IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml (hunk 6 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml]: Still here.

File: IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/DocProperty.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV" Name="TemperatureDuringInstallation" UniqueId="f9e5f6c5-dfbe-49c3-954b-9955571e14df" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/DocProperty.xml]: `IfcThermodynamicTemperatureMeasure` type?

File: IFC4x3/PropertyEnumerations/j/PEnum_JointRelativePosition.xml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_JointRelativePosition" Name="PEnum_JointRelativePosition" UniqueId="a83c1187-4f56-4bc6-ad43-39a30df74794">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/j/PEnum_JointRelativePosition.xml]: This looks familiar. Could you not generalize an enum that looks like this for example and reuse:
- `LEFT`
- `RIGHT`
- `MIDDLE`
- `UP`
- `DOWN`
and name it PEnum_RelativePosition?

File: IFC4x3/PropertyEnumerations/w/PEnum_WeldedJointRelativePosition.xml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_WeldedJointRelativePosition" Name="PEnum_WeldedJointRelativePosition" UniqueId="a83c1187-4f56-4bc6-ad43-39a30df74794">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/w/PEnum_WeldedJointRelativePosition.xml]: Same comment as above.

File: IFC4x3/PropertyEnumerations/c/PEnum_Created.xml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_Created" Name="PEnum_Created" UniqueId="96aa27c1-f3c3-48c0-b5c2-2c464b43583c">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/c/PEnum_Created.xml]: `IfcOwnerHistory`!

File: IFC4x3/Properties/i/IsWeldedInFactory_0aqg$9_hP5xeWnJrydF220/Documentation.md [hunk 17]
```diff
@@ -0,0 +1 @@
+Indicates if the rail is welded in factory (TRUE) or not (FALSE).
```
Comment (SergejMuhic) [inline IFC4x3/Properties/i/IsWeldedInFactory_0aqg$9_hP5xeWnJrydF220/Documentation.md]: Reuse `ConstructionMethod`?

**Review by SergejMuhic — COMMENTED** [review]
Don't know if intentional but some constants have documentation others do not.

File: IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml (hunk 2 above)
Comment (fhulin) [reply IFC4x3/Properties/c/ConstructionCompany_3q5el6dSLCw8hGDPwFONxo/DocProperty.xml]: Done. a14c639b6c4248c34ffadf5422b6425f466a4277

File: IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml (hunk 6 above)
Comment (fhulin) [reply IFC4x3/Properties/t/TrackEdgeId_3LYQ8lEh97fw__L8ZChIgF/DocProperty.xml]: Done here. a14c639b6c4248c34ffadf5422b6425f466a4277

File: IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml (hunk 7 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/e/EXTERNALCOMPANY_2KTrw1IIzBqxUGLKizr1oq.xml]: Done Here. a14c639b6c4248c34ffadf5422b6425f466a4277

File: IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml (hunk 8 above)
Comment (fhulin) [reply IFC4x3/PropertyConstants/i/INTERNALSTAFF_23JTDKB_5CNBaN6AkA1eoQ.xml]: Done. Here a14c639b6c4248c34ffadf5422b6425f466a4277

File: IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/Documentation.md [hunk 18]
```diff
@@ -0,0 +1 @@
+normalising working temperature / [°C]
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/Documentation.md]:
```suggestion
normalising working temperature
```

**Review by AlexBrad1eyCT — APPROVED** [review]
All Good

File: IFC4x3/PropertyEnumerations/r/PEnum_RelativePosition.xml [hunk 19]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_PositionInTrack" Name="PEnum_PositionInTrack" UniqueId="eed86143-41d2-40c4-b714-58afc9b7e145">
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_RelativePosition" Name="PEnum_RelativePosition" UniqueId="eed86143-41d2-40c4-b714-58afc9b7e145">
+	<Documentation>Enumeration of possible relative position value. The position of something is the place where it is in relation to other things.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/r/PEnum_RelativePosition.xml]:
```suggestion
	<Documentation>Enumeration of possible relative position values. The position of something is the place where it is in relation to other things.</Documentation>
```

File: IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/Documentation.md [hunk 20]
```diff
@@ -0,0 +1 @@
+normalising working temperature
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/TemperatureDuringInstallation_3vvVR5txv9mvLBcLLN7XJV/Documentation.md]:
```suggestion
Normalised working temperature.
```

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**